simulation/lowfi_simulation/src/main/native/include/lowfisim/SimulatorComponent.h

```diff
@@ -0,0 +1,29 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <string>
+
+namespace frc {
+namespace sim {
+namespace lowfi {
+
+class SimulatorComponent {
+ public:
+  virtual ~SimulatorComponent() {}
+
+  virtual bool IsWrapperInitialized() = 0;
```

Comment:
const-qualified?

Reply:
The existing hal wrapper classes are not const qualified.  I originally had that but it didn't compile and I didn't want to go down the rabbit hole of changing ALL of the getters in those classes to be const qualified, and it feels gross to do it on a single function, potentially only in the three files that (currently) need it

Reply:
Const correctness is not "gross". Every function that can be const-qualified should be. HAL functions (the ones with HALSIM_ prefixes) aren't part of classes so they can be invoked from const-qualfied methods. const qualification enforces that member variables of the class aren't modified in the function and that any member functions called are also const-qualified.

Reply:
I didn't mean const correctness is gross, I use it whenever possible.  I meant I didn't want to change 40 files and a hundred functions to do it right, and it feels inappropriate to just do it on the 3 functions that I would need for this.  

Whenever I do one of these PR's I try to minimize the impact of what gets added and changed

Reply:
If it requires furthering bad API design decisions though, I would prefer those be fixed in a separate PR first before more stuff is added.

Reply:
In progress. Will hit #1164 during it too